torch/csrc/jit/codegen/cuda/iter_visitor.h

```diff
@@ -49,9 +46,31 @@ class TORCH_CUDA_API IterVisitor : public OptOutDispatch {
   // These functions will start at outputs and propagate up through the DAG
   // to inputs based on depth first traversal. Next could be called on a node
   // multiple times.
-  virtual std::vector<Statement*> next(Statement* stmt);
-  virtual std::vector<Statement*> next(Expr* expr);
-  virtual std::vector<Statement*> next(Val* v);
+  virtual inline std::vector<Statement*> next(Statement* stmt) {
```

Comment:
Having both `virtual` and `inline` seems odd to me. I thought it's invalid, but it seems valid. https://www.geeksforgeeks.org/inline-virtual-function/.

Reply:
Reasonable criticism. I was playing around with it explicitly for: `but whenever called using the object (without reference or pointer) of that class, can be inlined` Though I doubt it made any noticeable difference and can remove it. IterVisitor was surprisingly slower than a simple recursive approach which I do want to take a closer look at sometime in the future.